### ISSUE TITLE
staggerDirection value not assignable with typescript

### ISSUE THREAD

**Issue by theonlydaleking** [framer/motion#399]
[in the example](https://codesandbox.io/s/framer-motion-side-menu-mx2rw?fontsize=14&module=%2Fsrc%2FExample.tsx) in Navigation.tsx, typescript is reporting: 

`type number is not assignable to to type -1 | 1` 

This is due to the way typescript now handles literals and type widening in objects (after 3.4 I think it was?). 

A couple of options: 

1. Change the example to use an interface for variants
2. Change the example to use `staggerDirection: -1 as -1` 
3. Update type declaration file to accept type number instead of -1 | 1
4. refactor staggerDirection to be something like `reverseStaggerDirection: boolean`

**Comment by nathanredblur** [comment]
I have similar issue when I move my variants to a variable so to quick fix this, I move it all inside the element.

I notice that the error happen when you use 'transition' attribute.

No issues
```
const itemSize = 23;
<motion.ul
      initial={false}
      animate={isOpen ? "open" : "closed"}
      custom={items.length + 1}
      ref={containerRef}
      variants={{
        open: (nItems) => ({
          overflow: 'hidden',
          height: `${itemSize * nItems}px`,
          transition: { staggerChildren: 0.07, delayChildren: 0.2 }
        }),
        closed: {
          overflow: 'hidden',
          height: 0,
          transition: { staggerChildren: 0.05, staggerDirection: -1, when: "afterChildren" }
        }
      }}>
```

issues
```
const itemSize = 23;
const variants = {
        open: (nItems) => ({
          overflow: 'hidden',
          height: `${itemSize * nItems}px`,
          transition: { staggerChildren: 0.07, delayChildren: 0.2 }
        }),
        closed: {
          overflow: 'hidden',
          height: 0,
          transition: { staggerChildren: 0.05, staggerDirection: -1, when: "afterChildren" }
        }
      }
<motion.ul
      initial={false}
      animate={isOpen ? "open" : "closed"}
      custom={items.length + 1}
      ref={containerRef}
      variants={variants}>
```

**Comment by dy-byron** [comment]
For those that want to keep your variants inside a variable: 

You can declare that variable as `Variants`

```
import { motion, Variants } from 'framer-motion'

...

const variants: Variants = {
  open: (nItems) => ({
    overflow: 'hidden',
    height: `${itemSize * nItems}px`,
    transition: { staggerChildren: 0.07, delayChildren: 0.2 }
  }),
  closed: {
    overflow: 'hidden',
    height: 0,
    transition: { staggerChildren: 0.05, staggerDirection: -1, when: "afterChildren" }
  }
}

<motion.ul
  initial={false}
  animate={isOpen ? "open" : "closed"}
  custom={items.length + 1}
  ref={containerRef}
  variants={variants}>

  ...

</motion.ul>
```

**Comment by InventingWithMonster** [comment]
@dy-byron Can you check again without explicit typing with version 1.7.2? This should be fixed now.

**Comment by dy-byron** [comment]
Indeed, it's fixed with version 1.7.2

Thanks @InventingWithMonster 🤙
